File: DesignPatternsExercises/java/SingletonDesignPattern/Exercise/Balance.java
class Balance {
	__________(1)__________ Balance firstInstance = null;
	private int balance;

	__________(2)__________ {
		this.balance = balance;
	}

	public static Balance getInstance() {
		if(__________(3)__________) {
			firstInstance = new Balance(100);
		}
		return firstInstance;
	}
	public void withdraw(int amount) {
	  if(balance > amount)
	  	balance -= amount;
	  else
	        System.out.println("Insufficient balance");
	}
	public void deposit(int amount){
		balance += amount;
	}
	public int getBalance(){
		return balance;
	}
}

File: DesignPatternsExercises/ruby/singleton_design_pattern/exercise/balance.rb
class Balance
  attr_reader __________(1)__________

  def __________(2)__________(balance)
    @balance = balance
    __________(3)__________ = nil
  end

  def __________(4)__________.instance
    @first_instance = __________(5)__________(100) if @first_instance.nil?
    __________(6)__________
  end

  def withdraw(amount)
    @balance > amount ? (@balance -= amount) : (puts 'Insufficient balance')
  end

  def deposit(amount)
    @balance += amount
  end
end
